clang/lib/CodeGen/CGOpenMPRuntime.cpp

```diff
@@ -1370,10 +1375,14 @@ llvm::Value *CGOpenMPRuntime::emitUpdateLocation(CodeGenFunction &CGF,
     SrcLocStr = OMPBuilder.getOrCreateDefaultSrcLocStr(SrcLocStrSize);
   } else {
     std::string FunctionName;
+    std::string FileName;
     if (const auto *FD = dyn_cast_or_null<FunctionDecl>(CGF.CurFuncDecl))
       FunctionName = FD->getQualifiedNameAsString();
     PresumedLoc PLoc = CGF.getContext().getSourceManager().getPresumedLoc(Loc);
-    const char *FileName = PLoc.getFilename();
+    if (CGF.getDebugInfo())
```

Comment:
I'm *fairly* sure that this check is redundant, because in this branch debug info should be available, but an extra null check seems safer than randomly crashing.

I considered replacing this with an assert and running it through clang's test suite, but my machine is rather slow :/

Reply:
And ran the full test suite, it is redundant. I'll replace it with an assertion.